src/script/lua_api/l_inventory.cpp

```diff
@@ -273,18 +270,17 @@ int InvRef::l_set_lists(lua_State *L)
 	}
 
 	// Make a temporary inventory in case reading fails
```

Comment:
It could fail if the inventory provided is invalid.  If you think it should just clear the inventory and try, I'll do that.